src/test/kotlin/uk/gov/justice/digital/hmpps/prisonregister/model/PrisonServiceTest.kt

```diff
@@ -0,0 +1,198 @@
+package uk.gov.justice.digital.hmpps.prisonregister.model
+
+import com.nhaarman.mockito_kotlin.eq
+import com.nhaarman.mockito_kotlin.mock
+import com.nhaarman.mockito_kotlin.verify
+import com.nhaarman.mockito_kotlin.verifyNoMoreInteractions
+import com.nhaarman.mockito_kotlin.whenever
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.assertThatThrownBy
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.mockito.ArgumentMatchers.anyString
+import uk.gov.justice.digital.hmpps.prisonregister.resource.PrisonDto
+import java.util.Optional
+import javax.persistence.EntityNotFoundException
+
+class PrisonServiceTest {
```

Comment:
PrisonService uses the Kotlin CrudRepository extension function findByIdOrNull. However this is a static function and cannot be mocked by Mockito - or mockito_kotlin.   Fortunately (really?) findByIdOrNull delegates to the CrudRepository method findById and so the tests below stub/mock the findById method instead of findByIdOrNull.